internal/provider/repository_branch_protection_resource.go

```diff
@@ -0,0 +1,816 @@
+	data.StatusCheckContexts = types.ListNull(types.StringType)
+	data.MergeWhitelistUsernames = types.ListNull(types.StringType)
+	data.MergeWhitelistTeams = types.ListNull(types.StringType)
+	data.ApprovalsWhitelistUsernames = types.ListNull(types.StringType)
+	data.ApprovalsWhitelistTeams = types.ListNull(types.StringType)
+
+	if len(protection.PushWhitelistUsernames) > 0 {
+		elements := make([]attr.Value, len(protection.PushWhitelistUsernames))
+		for i, username := range protection.PushWhitelistUsernames {
+			elements[i] = types.StringValue(username)
+		}
+		data.PushWhitelistUsernames, _ = types.ListValue(types.StringType, elements)
+	}
+
+	if len(protection.PushWhitelistTeams) > 0 {
+		elements := make([]attr.Value, len(protection.PushWhitelistTeams))
+		for i, team := range protection.PushWhitelistTeams {
+			elements[i] = types.StringValue(team)
+		}
+		data.PushWhitelistTeams, _ = types.ListValue(types.StringType, elements)
+	}
+
+	if len(protection.StatusCheckContexts) > 0 {
+		elements := make([]attr.Value, len(protection.StatusCheckContexts))
+		for i, checkContext := range protection.StatusCheckContexts {
+			elements[i] = types.StringValue(checkContext)
+		}
+		data.StatusCheckContexts, _ = types.ListValue(types.StringType, elements)
+	}
+
+	if len(protection.MergeWhitelistUsernames) > 0 {
+		elements := make([]attr.Value, len(protection.MergeWhitelistUsernames))
+		for i, username := range protection.MergeWhitelistUsernames {
+			elements[i] = types.StringValue(username)
+		}
+		data.MergeWhitelistUsernames, _ = types.ListValue(types.StringType, elements)
+	}
+
+	if len(protection.MergeWhitelistTeams) > 0 {
+		elements := make([]attr.Value, len(protection.MergeWhitelistTeams))
+		for i, team := range protection.MergeWhitelistTeams {
+			elements[i] = types.StringValue(team)
+		}
+		data.MergeWhitelistTeams, _ = types.ListValue(types.StringType, elements)
+	}
+
+	if len(protection.ApprovalsWhitelistUsernames) > 0 {
+		elements := make([]attr.Value, len(protection.ApprovalsWhitelistUsernames))
+		for i, username := range protection.ApprovalsWhitelistUsernames {
+			elements[i] = types.StringValue(username)
+		}
+		data.ApprovalsWhitelistUsernames, _ = types.ListValue(types.StringType, elements)
+	}
+
+	if len(protection.ApprovalsWhitelistTeams) > 0 {
+		elements := make([]attr.Value, len(protection.ApprovalsWhitelistTeams))
+		for i, team := range protection.ApprovalsWhitelistTeams {
+			elements[i] = types.StringValue(team)
+		}
+		data.ApprovalsWhitelistTeams, _ = types.ListValue(types.StringType, elements)
+	}
+
+	// Save data into Terraform state
+	diags := response.State.Set(ctx, &data)
+	response.Diagnostics.Append(diags...)
+}
+
+// Helper function to convert model to CreateBranchProtectionOption.
+func (r *repositoryBranchProtectionResource) modelToCreateOption(ctx context.Context, data *repositoryBranchProtectionResourceModel) forgejo.CreateBranchProtectionOption {
```

Comment:
Looks like you're not using the `ctx context.Context` parameter - this will trigger linter warnings.